content/src/entrypoints/run-maintenance.ts

```diff
@@ -0,0 +1,42 @@
+import { Lifecycle } from '@well-known-components/interfaces'
+import { createLogComponent } from '@well-known-components/logger'
+import { createTestMetricsComponent } from '@well-known-components/metrics'
+import path from 'path'
+import { EnvironmentBuilder, EnvironmentConfig } from '../Environment'
+import { deleteUnreferencedFiles } from '../logic/delete-unreferenced-files'
+import { metricsDeclaration } from '../metrics'
+import { migrateContentFolderStructure } from '../migrations/ContentFolderMigrationManager'
+import { MigrationManagerFactory } from '../migrations/MigrationManagerFactory'
+import { createFileSystemContentStorage } from '../ports/contentStorage/fileSystemContentStorage'
+import { createFsComponent } from '../ports/fs'
+import { createDatabaseComponent } from '../ports/postgres'
+import { MaintenanceComponents } from '../types'
+
+void Lifecycle.run({
+  async main(program: Lifecycle.EntryPointParameters<MaintenanceComponents>): Promise<void> {
+    const { components, startComponents, stop } = program
+
+    await components.migrationManager.run()
+
+    await migrateContentFolderStructure(components)
+
+    await startComponents()
+
+    await deleteUnreferencedFiles(components)
+
+    await stop()
+  },
+
+  async initComponents() {
+    const logs = createLogComponent()
+    const metrics = createTestMetricsComponent(metricsDeclaration)
```

Comment:
do we need metrics in maintenance mode?

Reply:
Nope.. but the db component need it for its creation :(. Should i create a fake one?